src/addons/model.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * @fileoverview A general modelling class used by the Presenter's
+ * getModel function.
+ */
+
+import { set, Emitter } from '../microcosm'
+
+function isObservable(binding) {
+  return binding && typeof binding.subscribe === 'function'
+}
+
+function isCallable(binding) {
+  return binding && typeof binding.call === 'function'
+}
+
+function invoke(binding, repo, scope) {
+  if (isCallable(binding)) {
+    return binding.call(scope, repo.state, repo)
+  }
+
+  return binding
+}
+
+export default class Model extends Emitter {
+  /**
+   * @param {Microcosm} repo Track this Microcosm instance for updates
+   * @param {scope} scope Scope to invoke functional bindings
+   */
+  constructor(repo, scope) {
+    super()
+
+    this.repo = repo
+    this.scope = scope
+    this.bindings = {}
+    this.subscriptions = {}
+    this.value = {}
+
+    this.repo.on('change', this.compute, this)
+  }
+
+  /**
+   * Track an observable. Sending updates to a given key.
+   * @param {string} key
+   * @param {Observable} observable
+   */
+  track(key, observable) {
+    let last = this.subscriptions[key]
+    let next = observable.subscribe(value => this.set(key, value))
+
+    this.subscriptions[key] = next
+
+    if (last) {
+      last.unsubscribe()
+    }
+  }
+
+  /**
+   * @param {Object} bindings An set of key/value pairs for building a model
+   */
+  bind(bindings) {
+    this.bindings = {}
+
+    for (var key in bindings) {
+      let binding = bindings[key]
+
+      if (isObservable(binding)) {
+        this.track(key, binding)
+      } else {
+        this.bindings[key] = binding
+      }
+    }
+
+    this.compute()
+  }
+
+  /**
+   * Update a specific model key. Emits a change event
+   * @param {string} key
+   * @param {*} value
+   */
+  set(key, value) {
+    let next = set(this.value, key, value)
+
+    if (this.value !== next) {
+      this.value = next
+      this._emit('change', this.value)
```

Comment:
That's correct. Observables operate on a single key, so we need to update that key as they change, and bubble that change up to the Presenter. 

So yes. This ends up with n+1 in the following case:

```
getModel () {
  return {
    a: state => state.prop,
    b: new Observable(observer => {
      this.repo.on('change', state => observer.next(state.otherProp)
    })
  }
}
```

In this case, we'd get two updates. One option would be to batch, sort of like what we do for Microcosm change events. The other would be to call `compute` when an observable changes, executing the function callbacks if state has changed.

Tough problem :).